process_sample_data.py

```diff
@@ -67,33 +67,41 @@ def _parseData(self):
         tidal_volume_marker = "TidalVol:"
         pressure_marker = "Pressurex10:"
 
+        previous_data = [None, None, math.inf]
```

Comment:
Is this the only place that `previous_data` is ever set? Looks like you're missing a reset down in the previous_data equality check branch.